elements/power.lua

```diff
@@ -160,33 +164,30 @@ local Update = function(self, event, unit)
 	end
 
 	power.disconnected = disconnected
-
+	power.tapped = tapped
+	
 	local ptype, ptoken, altR, altG, altB = UnitPowerType(unit)
-	local t = self.colors.power[ptoken]
+	local t = self.colors.power[ptoken or ptype or 'MANA']
 	local r, g, b
 	local atlas = power.atlas or t.atlas
-	local bg = power.bg
-
+	
 	if(power.useAtlas and atlas and displayType ~= ALTERNATE_POWER_INDEX) then
 		r, g, b = t[1], t[2], t[3]
 		power:SetStatusBarAtlas(atlas)
 		power:SetStatusBarColor(1, 1, 1)
-		power:GetStatusBarTexture():SetDesaturated(disconnected or tapped)
-
-		if(bg) then
-			bg:SetAtlas(atlas)
+		if(power.colorTapping or power.colorDisconnected) then
+			power:GetStatusBarTexture():SetDesaturated(disconnected or tapped)
 		end
 	else
 		power:SetStatusBarTexture(power.texture)
-		if(bg) then bg:SetTexture(bg.texture) end
-
 		if(power.colorTapping and tapped) then
 			t = self.colors.tapped
 		elseif(power.colorDisconnected and disconnected) then
 			t = self.colors.disconnected
 		elseif(displayType == ALTERNATE_POWER_INDEX and power.altPowerColor) then
 			t = power.altPowerColor
 		elseif(power.colorPower) then
+			t = self.colors.power[ptoken]
```

Comment:
The order of preference for the color if `.colorPower` is set is:
  1. `ptoken`
  2. `power.GetAlternativeColor`
  3. `altR, altG, altB`
  4. `ptype`

The order of preference for atlas is `ptoken`, `ptype`, or `'MANA'`.

Because of this I had to reset the color table here, I wanted to keep the current behavior for that option.